gym/lib/gym/runner.rb

```diff
@@ -11,26 +11,35 @@ def run
         clear_old_files
         build_app
       end
-      verify_archive
-
+      verify_archive if Gym.project.produces_archive?
       FileUtils.mkdir_p(File.expand_path(Gym.config[:output_directory]))
 
       if Gym.project.ios? || Gym.project.tvos?
         fix_generic_archive # See https://github.com/fastlane/fastlane/pull/4325
-        package_app
+        package_app if Gym.project.produces_archive?
         fix_package
         compress_and_move_dsym
-        path = move_ipa
+        path = move_ipa if Gym.project.produces_archive?
         move_manifest
         move_app_thinning
         move_app_thinning_size_report
         move_apps_folder
-
-        path
       elsif Gym.project.mac?
         compress_and_move_dsym
-        copy_mac_app
+        copy_mac_app if Gym.project.mac_app?
+        path = extract_files_from_archive("Products/usr/local/bin/*") if Gym.project.command_line_tool?
+        path = extract_files_from_archive("Products/usr/local/lib/*") if Gym.project.mac_library?
+        path = Gym.project.build_settings(key: "BUILT_PRODUCTS_DIR")
+      end
+
+      if Gym.project.library? || Gym.project.framework?
+        base_framework_path = Gym.project.build_settings(key: "PROJECT_TEMP_ROOT")
+        base_framework_path = Gym.config[:derived_data_path] + "/Build/Intermediates/" if Gym.config[:derived_data_path]
```

Comment:
ok this looks confusing, but during tests - if we do not set the "-derivedDataPath" on xcodebuild - then it defaults to something, `PROJECT_TEMP_ROOT`- defaults directly to the Intermediates, which are used for frameworks and librarys, as they cannot produce a xcarchive :(

but once set by the user with :drived_data_path, it come's in without '/Build/Intermediates'
if you have a better way to do it i am open for feedback.